### ISSUE TITLE
Keyerror

### ISSUE THREAD

**Issue by RollMediaDeveloper** [radiorabe/suisa_sendemeldung#521]
Tested containerized code with following script:
sudo podman run --rm -ti -e BEARER_TOKEN=token -e STREAM_ID=s-xxxxxxxx ghcr.io/radiorabe/suisasendemeldung:latest suisa_sendemeldung --project-id=xxxxx --timezone=Europe/Xxxxxxxx --file --filetype=xlsx

Scritp loads results from ACRCloud, but gives the following error:
  File "/usr/local/bin/suisa_sendemeldung", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/lib/python3.11/site-packages/suisa_sendemeldung/suisa_sendemeldung.py", line 865, in main
    data = merge_duplicates(data)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/suisa_sendemeldung/suisa_sendemeldung.py", line 427, in merge_duplicates
    prev["metadata"]["played_duration"]
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
KeyError: 'played_duration'

What causes error?

**Comment by hairmare** [comment]
Hi @RollMediaDeveloper 

Thanks for taking the time to post your issue here!

We seem to use the "custom stream" solution on ACRCloud and ours looks like this in [the dashboard](https://console.acrcloud.com/bm?region=eu-west-1#/custom):

![image](https://github.com/radiorabe/suisa_sendemeldung/assets/116588/2c8fd469-558a-42b3-a3fb-c6784ba6bd57)

Are you also using the broadcast monitoring product?

I'll probably also need a JSON dump from ACRCloud to debug this further. Are you comfortable with tools like `curl` to export some raw data from ACRCloud?

A potential fix might be to ignore the "played_duration" field if it's not available. Let's figure out why you don't have it first and then proceed from there!

Cheers,
Lucas

**Comment by RollMediaDeveloper** [comment]
Yes, we are using broadcast monitoring also. Attached is results exported with curl from one day; I don't see 'Played Duration' info there, but if I export results from Console, it is there.
[results080724.csv](https://github.com/user-attachments/files/16139903/results080724.csv)

**Comment by hairmare** [comment]
Thanks for the dump! I think i can narrow down the issue... Our bucket looks like this:
![image](https://github.com/radiorabe/suisa_sendemeldung/assets/116588/8386985e-cf25-4ae6-9e9a-146630f138ea)

it uses a "non-realtime for music" profile that is configured like so:

> | Config ID | Name | Length For Record | Interval | Record | Noise Level | Real-Time |
> | ---- | ---- | ---- | ---- | ---- | ---- | ---- | 
> | 1 | non-realtime for music | 10 | 0 | Non-Record | Antinoise-Low | Non-Realtime |

When i look you your JSON, it seems to be configured to use a real-time configuration.

```json
{
  "data": [
    {
      "metadata": {
        "type": "real-time",
```

> | Config ID | Name | Length For Record | Interval | Record | Noise Level | Real-Time |
> | ---- | ---- | ---- | ---- | ---- | ---- | ---- | 
> | 4 | realtime | 10 | 3 | Non-Record | Antinoise-Low | Realtime |

As far as i understand it the real-time profile is meant for building things like a "now playing" feature for a website, while the "non real-time" [profiles](https://console.acrcloud.com/bm?region=eu-west-1#/custom-configs) are for broadcast monitoring and reporting. 

The "non real-time" profiles take a while until they return data, we usually get data some 10-20 minutes after we play a track. For our purpose the trade-off was that we get more concise reporting data from acrcloud.

Can you try if a "non real-time" configuration makes the `played_duration` field appear in your json dumps?

**Comment by RollMediaDeveloper** [comment]
[results150724.csv](https://github.com/user-attachments/files/16249367/results150724.csv)
Changed configuration to 'non-realtime', but I got same errors with script. However exported results with curl now show 'played_duration' info.

**Comment by hairmare** [comment]
I've not been able to reproduce the error with [results150724.csv](https://github.com/user-attachments/files/16249367/results150724.csv).

Maybe it doesn't work because the code tries to download results140724.csv at the same time. The code does this if the local timezone timezone does not match UTC. Can you provide a file for 2024-07-14 to help me verify this?

Also, do you still get the error if you try exporting a later date?

**Comment by hairmare** [comment]
We've ran into several KeyErrors of our own as of lately. Those have been fixed and they might also have fixed the issues reported here.

I'm going to go ahead an close this issue, there are currently a few 1.0.0-alpha pre-release images, if those releases are still affected by the issue now would be the perfect time to report the issues so we can include potential fixes in 1.x.
